solutions/silver/ac-nuske.mdx

```diff
@@ -283,4 +283,76 @@ public class Main {
 ```
 
 </JavaSection>
+
+<PySection>
+
+```py
+MAX_SIZE = 2000
+
+n, m, q = map(int, input().split())
+
+grid = [[False] * (MAX_SIZE + 1) for _ in range(MAX_SIZE + 1)]
+
+# Read the grid
+for i in range(1, n + 1):
+	row = input()
+	for j in range(1, m + 1):
+		grid[i][j] = row[j - 1] == "1"
+
+# Overall prefix sums array. pref[x][y] is how many components
+# we have in the 2D array from (1, 1) to (x, y), inclusive.
+pref = [[0] * (MAX_SIZE + 1) for _ in range(MAX_SIZE + 1)]
+
+for i in range(1, n + 1):
+	for j in range(1, m + 1):
+		pref[i][j] = pref[i - 1][j] + pref[i][j - 1] - pref[i - 1][j - 1]
+		if grid[i][j]:
+			"""
+			  0
+			0 1
+			A new connected component forms!
+			"""
+			if not grid[i - 1][j] and not grid[i][j - 1]:
+				pref[i][j] += 1
+
+			"""
+              1
+            1 1
+            Two connected componets merge!
```

Comment:
bad formatting

Reply:
bot is to blame

Reply:
corrected it.

Reply:
> bot is to blame

this wouldn't happen if you used tabs to indent ur code like the guide convention states

Reply:
but i do use tabs

Reply:
a look at your initial commit shows that you clearly didn't